### PR TITLE
workerutil: Add SetLogContents to store

### PR DESCRIPTION
**Pull request by efritz**
This PR adds a required `log_contents` field to all tables that describe a job consumed by an internal/worker process, and a method to set that field during its processing.

This closes https://github.com/sourcegraph/sourcegraph/issues/14831.

### CODE REVIEW

**Comment by efritz** [comment]
@sourcegraph/campaigns Could you please determine if I've broken the reconciler here? Tests are currently passing but I'm incredulous about it (maybe I only needed to update the repo-updater sync worker because it contained a `SELECT *` and had an associated view over that table).

File: internal/workerutil/dbworker/store/store.go [hunk 1]
```diff
@@ -386,6 +391,23 @@ SET {state} = 'queued', {process_after} = %s
...
+	return s.Exec(ctx, s.formatQuery(
+		setLogContentsQuery,
+		quote(s.options.TableName),
+		logContents,
+		id,
+	))
+}
+
+const setLogContentsQuery = `
+-- source: internal/workerutil/store.go:Requeue
```
Comment (eseliger) [inline internal/workerutil/dbworker/store/store.go:405]:
```suggestion
-- source: internal/workerutil/store.go:SetLogContents
```

**Review by eseliger — COMMENTED** [review]
Doesn't the `changesets` table need this field then as well? 

At least the comment states

> 	// The target table (and the target view referenced by `ViewName`) must have the following columns and types: [...] log_contents

File: cmd/repo-updater/repos/store.go [hunk 2]
```diff
@@ -1168,6 +1186,7 @@ func scanJobs(rows *sql.Rows) ([]SyncJob, error) {
 			&job.ProcessAfter,
 			&job.NumResets,
 			&job.NumFailures,
+			&logContents,
```
Comment (mrnugget) [inline cmd/repo-updater/repos/store.go:1189]: Are you sure you want a pointer to a pointer here?

**Review by mrnugget — APPROVED** [review]
I don't think you broke anything — it would only crash if we were to call `SetLogContents`, right?

**Comment by efritz** [comment]
@eseliger @mrnugget Will add the column to changesets table as well.

File: cmd/repo-updater/repos/store.go (hunk 2 above)
Comment (efritz) [reply cmd/repo-updater/repos/store.go:1189]: It must be otherwise you get `destination pointer is nil`.

**Pull request merged**